### PR TITLE
[DEVHAS-223] Set revision in componentStub if set in CDQ's spec

### PR DESCRIPTION
**Pull request by johnmcollier**
Signed-off-by: John Collier <jcollier@redhat.com>

### What does this PR do?:
Updates the componentStub returned by the CDQ to include the Git revision if `GitSource.revision` was set in the CDQ's spec. 

### Which issue(s)/story(ies) does this PR fixes:
https://issues.redhat.com/browse/DEVHAS-223

### PR acceptance criteria:
Tests pass and ComponentStub now includes the proper Git revision if set in the CDQ spec.

### CODE REVIEW

**Comment by yangcao77** [comment]
Also need to update https://github.com/redhat-appstudio/application-service/blob/ab24f58d9d38ae0e59e69b5b7291ab7a80e1ab79/controllers/componentdetectionquery_controller.go#L148

File: controllers/componentdetectionquery_controller.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 			} else {
 				// Use SPI to retrieve the devfile from the private repository
 				// TODO - maysunfaisal also search for Dockerfile
-				devfileBytes, err = spi.DownloadDevfileUsingSPI(r.SPIClient, ctx, componentDetectionQuery.Namespace, source.URL, "main", context)
+				devfileBytes, err = spi.DownloadDevfileUsingSPI(r.SPIClient, ctx, componentDetectionQuery.Namespace, source.URL, source.Revision, context)
```
Comment (yangcao77) [inline controllers/componentdetectionquery_controller.go:148]: I'm wondering if no `Revision` has been passed in, will this cause any error?
we might want to add more unit tests on `Revision =""`,  and `Revision=<other than main>`

Comment (johnmcollier) [reply controllers/componentdetectionquery_controller.go:148]: Yes good find. The problem is that the actual SPI client requires a running Kube cluster (with SPI installed) in order to function properly, so it's not something that we can really test in the controller tests.

Comment (johnmcollier) [reply controllers/componentdetectionquery_controller.go]: Updated.

**Review by yangcao77 — APPROVED** [review]
changes look good to me

**Pull request merged**